src/crab/splitdbm/graph_views.hpp

```
@@ -0,0 +1,438 @@
+        return g.elem(perm[x], perm[y]);
+    }
+
+    const Weight* lookup(const VertId x, const VertId y) const {
+        if (perm[x] >= g.size() || perm[y] >= g.size()) {
+            return nullptr;
+        }
+        return g.lookup(perm[x], perm[y]);
+    }
+
+    // Precondition: elem(x, y) is true.
+    Weight edge_val(const VertId x, const VertId y) const {
```

Clang (14.0.6)

[warning] function 'edge_val' should be marked [[nodiscard]]

(modernize-use-nodiscard)